rpc/internal/test_results_service.go

```diff
@@ -83,16 +84,63 @@ func (s *testResultsService) AddTestResults(ctx context.Context, results *TestRe
 	}
 
 	if !record.Info.HistoricalDataDisabled {
-		for _, res := range exportedResults {
-			// TODO: Remove this job with EVG-14440.
-			j := units.NewHistoricalTestDataJob(s.env, record.Info, res)
-			j.Run(ctx)
-			grip.Error(message.WrapError(j.Error(), message.Fields{
-				"message":     "failed to run historical test data job",
-				"info":        record.Info,
-				"test_result": res,
-			}))
-		}
+		go func() {
+			defer func() {
+				if err := recovery.HandlePanicWithError(recover(), nil, "historical test data update"); err != nil {
+					grip.Error(message.WrapError(err, message.Fields{
+						"message":           "failed to update historical test data",
+						"test_results_info": record.Info,
+					}))
+				}
+			}()
+
+			htdCtx, cancel := context.WithTimeout(context.Background(), time.Minute)
+			defer cancel()
+
+			conf := model.NewCedarConfig(s.env)
+			if err := conf.Find(); err != nil {
+				grip.Error(message.WrapError(errors.Wrap(err, "finding cedar configuration"), message.Fields{
+					"message":           "failed to update historical test data",
+					"test_results_info": record.Info,
+				}))
+				return
+			}
+			if conf.Flags.DisableHistoricalTestData {
+				return
+			}
+
+			taskName := record.Info.DisplayTaskName
+			if taskName == "" {
+				taskName = record.Info.TaskName
+			}
```

Comment:
Since we've done this [task name logic before](https://github.com/evergreen-ci/cedar/blob/997a0eeb6ba85cbf227ca91b427e3d476a296afa/rpc/internal/test_results.ext.go#L17-L20), can this task name logic be a method on `TestResultsInfo`, like `HistoricalTaskName()` or something?

Reply:
The other place you referenced is on a different data type (the protobuf's `TestResults` struct, not the `model.TestResults` struct) so it wouldn't actually reduce duplicate code.